src/main/java/com/example/kidz_school/service/impl/UserServiceImpl.java

```diff
@@ -68,6 +70,84 @@ public void resetPassword(String email, String newPassword) {
+                !phoneIsValid(userRequest.getPhone(), model)) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void sendVerificationMassage(User user, UUID uuid) {
+        String messageBody = new StringBuilder().append("Hi ")
+                .append(user.getFirstName())
+                .append("\n Please verify your email by clicking on this url: ")
+                .append(siteUrlConfig.getUrl())
+                .append("/sign-up-or-login/verify?token=")
+                .append(uuid).toString();
+        emailService.sendEmail(user.getEmail(), "Welcome", messageBody);
+    }
+
+    private boolean emailIsValid(String email, Model model) {
+        String emailRegex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,6}$";
+        if (!Pattern.compile(emailRegex).matcher(email).matches()) {
+            model.addAttribute("field", "email");
+            return false;
+        }
+        return true;
+    }
+
+    private boolean passwordIsValid(String password, String rePassword, Model model) {
+        String passwordRegex = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[@#$%^&+=()])(?=.*[a-zA-Z0-9]).{6,}$";
+        if (!Pattern.compile(passwordRegex).matcher(password).matches()) {
+            model.addAttribute("field", "password");
+            model.addAttribute("message", "Password must include at least one uppercase, one lowercase, one digit, one special character, and be at least 6 characters long");
+            return false;
+        }
+        if (!password.equals(rePassword)) {
+            model.addAttribute("field", "re-password");
+            model.addAttribute("message", "Re-password mismatch.");
+            return false;
+        }
+        return true;
+    }
+
+    private boolean nameIsValid(String name, String fieldName, Model model) {
+        if (name.length() < 3 || name.length() > 20) {
+            model.addAttribute("field", fieldName);
+            model.addAttribute("message", fieldName + " must not be empty, must have more than 3 characters, and less than 10 characters");
+            return false;
+        }
+        return true;
+    }
+    private boolean lastnameIsValid(String lastname, String fieldName, Model model) {
+        if (lastname.length() <= 3 || lastname.length() >= 20) {
+            model.addAttribute("field", fieldName);
+            model.addAttribute("message", fieldName + " must not be empty, must have more than 3 characters, and less than 10 characters");
+            return false;
+        }
+        return true;
+    }
+
+    private boolean phoneIsValid(String phone, Model model) {
+        String phoneRegex = "^\\+\\d{1,3}( )\\d{2}( )\\d{6}$";
+        if (!Pattern.compile(phoneRegex).matcher(phone).matches()) {
+            model.addAttribute("field", "Phone");
+            model.addAttribute("message", "Phone number must have the following format: +XXX XX XXXXXX");
+            return false;
+        }
+        return true;
+    }
```

Comment:
But even if you're validating like this (which is wrong and you must change it), how can you put all the code in the service. Were is the Single Responsibility?